options/validate.go

```diff
@@ -17,18 +17,41 @@ type FilteringOption struct {
 	Deny      []QueryValidate_FilterOperator
 }
 
+func getFieldInfo(path []string, messageInfo map[string]FilteringOption) (FilteringOption, error) {
```

Comment:
I would propose a simple algorithm which doesn't have nesting level limitations:
- if there is no tag found in messageInfo:
  - for each tag in messageInfo
    - if it's a * notation(the last symbol is *)
      - if fieldTag starts with a prefix of base prefix of tag(without a star)
        - return messageInfo[tag]
  - return err(unknown field)